docs/working-with-cdk-for-terraform/terraform-assets.md

```diff
@@ -0,0 +1,47 @@
+# Terraform Assets
+
+A Terraform Asset takes a file or directory outside of the CDK for Terraform context and moves it into it. Assets copy referenced files into the stacks context for further usage in other resources.
+
+This is useful for:
+
+- Lambda functions for copying over previously generated zip files
+- S3 Content for deploying static local files
+
+```typescript
+import * as path from "path";
+import { Construct } from "constructs";
+import { App, TerraformStack, TerraformAsset, AssetType } from "cdktf";
+import { AwsProvider } from "./.gen/providers/aws/aws-provider";
+import { S3BucketObject } from "./.gen/providers/aws/s3-bucket-object";
+import { S3Bucket } from "./.gen/providers/aws/s3-bucket";
+
+
+class MyStack extends TerraformStack {
+  constructor(scope: Construct, name: string) {
+    super(scope, name);
+
+    new AwsProvider(this, "provider", {
+      region: "us-west-2",
+    });
+
+    const bucket = new S3Bucket(this, "bucket", {
+      bucket: "demo",
+    });
+
+    const asset = new TerraformAsset(this, "lambda-asset", {
+      path: path.resolve(__dirname, "../lambda.zip") // can also point to a folder
+    });
+
+    new S3BucketObject(this, "lambda-archive", {
+      bucket: bucket.bucket,
+      key: "lambda.zip",
+      source: asset.path,
```

Comment:
Would be neat if there was something like `asset.fileName` if the path represents a file

Reply:
Just to be sure, you mean as an additional property not instead of `path`, right?

Reply:
yes, in addition to path. Essentially to use if as `key` in this example.